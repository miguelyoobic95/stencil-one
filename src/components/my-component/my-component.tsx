import { Component, Prop, h, JSX, Event, EventEmitter } from '@stencil/core';
import { format } from '../../utils/utils';
import { Gsap } from '../../utils/gsap';
@Component({
  tag: 'my-component',
  styleUrl: 'my-component.css',
  shadow: true
})
export class MyComponent {
  /**
   * The first name
   */
  @Prop() first: string;

  /**
   * The middle name
   */
  @Prop() middle: string;

  /**
   * The last name
   */
  @Prop() last: string;

  @Event() buttonClicked: EventEmitter<string>;
  private getText(): string {
    return format(this.first, this.middle, this.last);
  }
  onClicked(value: string) {
    console.log('Value Clicked is ', value);
  }
  gsap: Gsap;

  render() : JSX.Element {
    this.gsap = new Gsap();
    return (
      <div class="nice">
      <span>
        Hello, World! I'm {this.getText()}
      </span>
      <button onClick={()=>this.onClicked('ABCD')}>
        Click Me!
      </button>
      </div>
    );
  }
}
