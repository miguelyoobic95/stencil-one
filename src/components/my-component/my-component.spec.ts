import {newSpecPage} from '@stencil/core/testing';

import {MyComponent} from './my-component';

import { serializer } from '../../../test/stencil-serializer';

expect.addSnapshotSerializer(serializer);
it('Should render', async() => {
  const page = await newSpecPage({
    components: [MyComponent],
    html: `<my-component first="Hello" last="World"></my-component>`,
    serializedShadowDom: true
  });

  // expect(root).toEqualHtml(`
  //   <my-component class=\"hydrated\" first=\"Hello\" last=\"World\">
  //   <shadow-root>
  //     <div class=\"nice\">
  //       <span>
  //         Hello, World! I'm Hello World
  //       </span>
  //       <button>
  //         Click Me!
  //       </button>
  //     </div>
  //   </shadow-root>
  // </my-component>
  // `);

  expect(page.doc).toMatchSnapshot();
  let text = page.root.shadowRoot.querySelector('span');
  expect(text.textContent).toBe(`Hello, World! I'm Hello World`);
  expect(page.root.first).toEqual('Hello');
});

it('Should emit', async() => {
  const {root, win} = await newSpecPage({
    components: [MyComponent],
    html: `<my-component first="John" last="Doe"></my-component>`
  });
  let button = root.shadowRoot.querySelector('button');
  let buttonClicked = jest.fn();
  win.addEventListener('buttonClicked', buttonClicked);
  button.click();
  expect(buttonClicked).toHaveBeenCalled;
});

it('Should emit', async() => {
  const {root, win} = await newSpecPage({
    components: [MyComponent],
    html: `<my-component first="John" last="Doe"></my-component>`
  });
  let button = root.shadowRoot.querySelector('button');
  let buttonClicked = jest.fn();
  win.addEventListener('buttonClicked', buttonClicked);
  button.click();
  expect(buttonClicked).toHaveBeenCalled;
});
