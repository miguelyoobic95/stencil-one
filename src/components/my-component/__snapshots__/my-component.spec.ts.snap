// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
MockDocumentFragment {
  "childNodes": Array [
    MockElement {
      "childNodes": Array [
        MockElement {
          "childNodes": Array [
            MockTextNode {
              "childNodes": Array [],
              "nodeName": "#text",
              "nodeType": 3,
              "nodeValue": "Hello, World! I'm Hello World",
              "ownerDocument": MockDocument {
                "childNodes": Array [
                  MockDocumentTypeNode {
                    "childNodes": Array [],
                    "namespaceURI": null,
                    "nodeName": "!DOCTYPE",
                    "nodeType": 10,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                  MockElement {
                    "childNodes": Array [
                      MockElement {
                        "childNodes": Array [
                          MockElement {
                            "childNodes": Array [
                              MockTextNode {
                                "childNodes": Array [],
                                "nodeName": "#text",
                                "nodeType": 3,
                                "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                                "ownerDocument": [Circular],
                                "parentNode": [Circular],
                              },
                            ],
                            "namespaceURI": null,
                            "nodeName": "STYLE",
                            "nodeType": 1,
                            "nodeValue": null,
                            "ownerDocument": [Circular],
                            "parentNode": [Circular],
                          },
                        ],
                        "namespaceURI": null,
                        "nodeName": "HEAD",
                        "nodeType": 1,
                        "nodeValue": null,
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                      MockElement {
                        "childNodes": Array [
                          ,
                        ],
                        "namespaceURI": null,
                        "nodeName": "BODY",
                        "nodeType": 1,
                        "nodeValue": null,
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                    ],
                    "namespaceURI": null,
                    "nodeName": "HTML",
                    "nodeType": 1,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                ],
                "cookie": "",
                "defaultView": [MockWindow],
                "nodeName": "#document",
                "nodeType": 9,
                "nodeValue": null,
                "ownerDocument": null,
                "parentNode": null,
                "referrer": "",
              },
              "parentNode": [Circular],
              "s-hn": "my-component",
            },
          ],
          "namespaceURI": null,
          "nodeName": "SPAN",
          "nodeType": 1,
          "nodeValue": null,
          "ownerDocument": MockDocument {
            "childNodes": Array [
              MockDocumentTypeNode {
                "childNodes": Array [],
                "namespaceURI": null,
                "nodeName": "!DOCTYPE",
                "nodeType": 10,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
              MockElement {
                "childNodes": Array [
                  MockElement {
                    "childNodes": Array [
                      MockElement {
                        "childNodes": Array [
                          MockTextNode {
                            "childNodes": Array [],
                            "nodeName": "#text",
                            "nodeType": 3,
                            "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                            "ownerDocument": [Circular],
                            "parentNode": [Circular],
                          },
                        ],
                        "namespaceURI": null,
                        "nodeName": "STYLE",
                        "nodeType": 1,
                        "nodeValue": null,
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                    ],
                    "namespaceURI": null,
                    "nodeName": "HEAD",
                    "nodeType": 1,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                  MockElement {
                    "childNodes": Array [
                      ,
                    ],
                    "namespaceURI": null,
                    "nodeName": "BODY",
                    "nodeType": 1,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                ],
                "namespaceURI": null,
                "nodeName": "HTML",
                "nodeType": 1,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
            ],
            "cookie": "",
            "defaultView": [MockWindow],
            "nodeName": "#document",
            "nodeType": 9,
            "nodeValue": null,
            "ownerDocument": null,
            "parentNode": null,
            "referrer": "",
          },
          "parentNode": [Circular],
          "s-hn": "my-component",
        },
        MockButtonElement {
          "__listeners": Array [
            MockEventListener {
              "handler": [Function],
              "type": "click",
            },
          ],
          "childNodes": Array [
            MockTextNode {
              "childNodes": Array [],
              "nodeName": "#text",
              "nodeType": 3,
              "nodeValue": "Click Me!",
              "ownerDocument": MockDocument {
                "childNodes": Array [
                  MockDocumentTypeNode {
                    "childNodes": Array [],
                    "namespaceURI": null,
                    "nodeName": "!DOCTYPE",
                    "nodeType": 10,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                  MockElement {
                    "childNodes": Array [
                      MockElement {
                        "childNodes": Array [
                          MockElement {
                            "childNodes": Array [
                              MockTextNode {
                                "childNodes": Array [],
                                "nodeName": "#text",
                                "nodeType": 3,
                                "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                                "ownerDocument": [Circular],
                                "parentNode": [Circular],
                              },
                            ],
                            "namespaceURI": null,
                            "nodeName": "STYLE",
                            "nodeType": 1,
                            "nodeValue": null,
                            "ownerDocument": [Circular],
                            "parentNode": [Circular],
                          },
                        ],
                        "namespaceURI": null,
                        "nodeName": "HEAD",
                        "nodeType": 1,
                        "nodeValue": null,
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                      MockElement {
                        "childNodes": Array [
                          ,
                        ],
                        "namespaceURI": null,
                        "nodeName": "BODY",
                        "nodeType": 1,
                        "nodeValue": null,
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                    ],
                    "namespaceURI": null,
                    "nodeName": "HTML",
                    "nodeType": 1,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                ],
                "cookie": "",
                "defaultView": [MockWindow],
                "nodeName": "#document",
                "nodeType": 9,
                "nodeValue": null,
                "ownerDocument": null,
                "parentNode": null,
                "referrer": "",
              },
              "parentNode": [Circular],
              "s-hn": "my-component",
            },
          ],
          "namespaceURI": null,
          "nodeName": "BUTTON",
          "nodeType": 1,
          "nodeValue": null,
          "ownerDocument": MockDocument {
            "childNodes": Array [
              MockDocumentTypeNode {
                "childNodes": Array [],
                "namespaceURI": null,
                "nodeName": "!DOCTYPE",
                "nodeType": 10,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
              MockElement {
                "childNodes": Array [
                  MockElement {
                    "childNodes": Array [
                      MockElement {
                        "childNodes": Array [
                          MockTextNode {
                            "childNodes": Array [],
                            "nodeName": "#text",
                            "nodeType": 3,
                            "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                            "ownerDocument": [Circular],
                            "parentNode": [Circular],
                          },
                        ],
                        "namespaceURI": null,
                        "nodeName": "STYLE",
                        "nodeType": 1,
                        "nodeValue": null,
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                    ],
                    "namespaceURI": null,
                    "nodeName": "HEAD",
                    "nodeType": 1,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                  MockElement {
                    "childNodes": Array [
                      ,
                    ],
                    "namespaceURI": null,
                    "nodeName": "BODY",
                    "nodeType": 1,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                ],
                "namespaceURI": null,
                "nodeName": "HTML",
                "nodeType": 1,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
            ],
            "cookie": "",
            "defaultView": [MockWindow],
            "nodeName": "#document",
            "nodeType": 9,
            "nodeValue": null,
            "ownerDocument": null,
            "parentNode": null,
            "referrer": "",
          },
          "parentNode": [Circular],
          "s-hn": "my-component",
        },
      ],
      "namespaceURI": null,
      "nodeName": "DIV",
      "nodeType": 1,
      "nodeValue": null,
      "ownerDocument": MockDocument {
        "childNodes": Array [
          MockDocumentTypeNode {
            "childNodes": Array [],
            "namespaceURI": null,
            "nodeName": "!DOCTYPE",
            "nodeType": 10,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
          MockElement {
            "childNodes": Array [
              MockElement {
                "childNodes": Array [
                  MockElement {
                    "childNodes": Array [
                      MockTextNode {
                        "childNodes": Array [],
                        "nodeName": "#text",
                        "nodeType": 3,
                        "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                        "ownerDocument": [Circular],
                        "parentNode": [Circular],
                      },
                    ],
                    "namespaceURI": null,
                    "nodeName": "STYLE",
                    "nodeType": 1,
                    "nodeValue": null,
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                ],
                "namespaceURI": null,
                "nodeName": "HEAD",
                "nodeType": 1,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
              MockElement {
                "childNodes": Array [
                  ,
                ],
                "namespaceURI": null,
                "nodeName": "BODY",
                "nodeType": 1,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
            ],
            "namespaceURI": null,
            "nodeName": "HTML",
            "nodeType": 1,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
        ],
        "cookie": "",
        "defaultView": [MockWindow],
        "nodeName": "#document",
        "nodeType": 9,
        "nodeValue": null,
        "ownerDocument": null,
        "parentNode": null,
        "referrer": "",
      },
      "parentNode": [Circular],
      "s-hn": "my-component",
    },
  ],
  "host": ,
  "namespaceURI": null,
  "nodeName": "#document-fragment",
  "nodeType": 11,
  "nodeValue": null,
  "ownerDocument": MockDocument {
    "childNodes": Array [
      MockDocumentTypeNode {
        "childNodes": Array [],
        "namespaceURI": null,
        "nodeName": "!DOCTYPE",
        "nodeType": 10,
        "nodeValue": null,
        "ownerDocument": [Circular],
        "parentNode": [Circular],
      },
      MockElement {
        "childNodes": Array [
          MockElement {
            "childNodes": Array [
              MockElement {
                "childNodes": Array [
                  MockTextNode {
                    "childNodes": Array [],
                    "nodeName": "#text",
                    "nodeType": 3,
                    "nodeValue": "my-component{visibility:hidden}.hydrated{visibility:inherit}",
                    "ownerDocument": [Circular],
                    "parentNode": [Circular],
                  },
                ],
                "namespaceURI": null,
                "nodeName": "STYLE",
                "nodeType": 1,
                "nodeValue": null,
                "ownerDocument": [Circular],
                "parentNode": [Circular],
              },
            ],
            "namespaceURI": null,
            "nodeName": "HEAD",
            "nodeType": 1,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
          MockElement {
            "childNodes": Array [
              ,
            ],
            "namespaceURI": null,
            "nodeName": "BODY",
            "nodeType": 1,
            "nodeValue": null,
            "ownerDocument": [Circular],
            "parentNode": [Circular],
          },
        ],
        "namespaceURI": null,
        "nodeName": "HTML",
        "nodeType": 1,
        "nodeValue": null,
        "ownerDocument": [Circular],
        "parentNode": [Circular],
      },
    ],
    "cookie": "",
    "defaultView": [MockWindow],
    "nodeName": "#document",
    "nodeType": 9,
    "nodeValue": null,
    "ownerDocument": null,
    "parentNode": null,
    "referrer": "",
  },
  "parentNode": null,
}
`;
